xbmc/music/MusicDatabase.cpp

```diff
@@ -9032,86 +9032,40 @@ bool CMusicDatabase::GetBoxsetDiscs(const std::string& strBaseDir, CFileItemList
-    idAlbum = -1;
-  unsigned int end = strBaseDir.find("/", start + 9); // 9 is 'boxsets/' + 1
-  std::string tmp = strBaseDir.substr(start + 8, end - 18); // 8 = 'boxsets/', 18 = 'musicdb://boxsets/'
+    return false;
+
+  end = strBaseDir.find("/", start + 9); // 9 is 'boxsets/' + 1
+  tmp = strBaseDir.substr(start + 8, end - 18); // 8 = 'boxsets/', 18 = 'musicdb://boxsets/'
   idAlbum = atoi(tmp.c_str());
   start = end + 1; // set start to end of albumID field + '/'
   end = strBaseDir.find("/", start + 1);
-  std::string strDiscName = strBaseDir.substr(start, (strBaseDir.length() -1) - start);
+  strDiscName = strBaseDir.substr(start, (strBaseDir.length() -1) - start);
   if (idAlbum == -1)
     return false;
-  if (m_pDB.get() == NULL || m_pDS.get() == NULL)
-    return false;
-
-  try
-  {
-    int total = -1;
-    std::string strSQL = PrepareSQL("SELECT songview.* from songview WHERE songview.strDiscSubtitle = '%s' AND songview.idAlbum = %i ", strDiscName.c_str(), idAlbum) ;
-    SortDescription sorting;
-    CLog::Log(LOGDEBUG, "%s query = %s", __FUNCTION__, strSQL.c_str());
-    // run query
-    if (!m_pDS->query(strSQL))
-      return false;
-
-    int iRowsFound = m_pDS->num_rows();
-    if (iRowsFound == 0)
-    {
-      m_pDS->close();
-      return true;
-    }
-
-    // store the total value of items as a property
-    if (total < iRowsFound)
-      total = iRowsFound;
-    items.SetProperty("total", total);
-
-    DatabaseResults results;
-    results.reserve(iRowsFound);
-    if (!SortUtils::SortFromDataset(sorting, MediaTypeSong, m_pDS, results))
-      return false;
 
-    // get data from returned rows
-    items.Reserve(results.size());
-    const dbiplus::query_data &data = m_pDS->get_result_set().records;
-    int count = 0;
-    for (const auto &i : results)
-    {
-      unsigned int targetRow = (unsigned int)i.at(FieldRow).asInteger();
-      const dbiplus::sql_record* const record = data.at(targetRow);
-
-      try
-      {
-        CFileItemPtr item(new CFileItem);
-        GetFileItemFromDataset(record, item.get(), musicUrl);
-        // HACK for sorting by database returned order
-        item->m_iprogramCount = ++count;
-        items.Add(item);
-      }
-      catch (...)
-      {
-        m_pDS->close();
-        CLog::Log(LOGERROR, "%s: out of memory loading query: %s", __FUNCTION__, filter.where.c_str());
-        return (items.Size() > 0);
-      }
-    }
-
-    // cleanup
-    m_pDS->close();
-    return true;
-  }
-  catch (...)
-  {
-    // cleanup
-    m_pDS->close();
-    CLog::Log(LOGERROR, "%s(%s) failed", __FUNCTION__, filter.where.c_str());
-    return false;
-  }
+  buildUrl = "musicdb://boxsets/";
+  buildUrl += StringUtils::Format("%i/", idAlbum);
+  buildUrl += StringUtils::Format("%s/", strDiscName);
+  musicUrl.FromString(buildUrl);
+  musicUrl.AddOption("disctitle", strDiscName.c_str());
+  GetFilter(musicUrl, filter, sorting);
+  GetSongsByWhere(musicUrl.ToString(), filter, items);
```

Comment:
OK, but not what I meant. You don't need this method at all with its bespoke path parsing
Filter using idAlbum and discnumber not disc title which may not be unique.  See https://github.com/DaveTBlake/xbmc/blob/20b51ce9ddbbd7db2655771e73dd15717e21051b/xbmc/music/MusicDatabase.cpp#L11184-L11202
Where you call GetBoxsetDiscSongs is just getting songs with that filter which can already be on the item path

Reply:
I know disc title may not be unique.  Certainly in the case of auto-generated names it won't be, but the combination of idAlbum and disc title will be.  Maybe discnumber is indeed better though.  I've only quickly had a look at your code.  I've built it and tested it against my small sample set and it works fine, although you seem to have broken the disc name that I added to estuary.  No biggie though.

Reply:
Oh, on the subject of what is displayed, there _is_ a bit of an issue when we stack the albums and set pItem->SetLabel(strDiscSubtitle).  In estuary (haven't tested with confluence yet, but will do) only two views out of the four show the correct disc title.  Shift and WideList both show correctly, Wall and InfoWall both show the album title, not the disc title.  I don't yet know if that's a skin thing or a core thing but I thought it might be worth mentioning as whichever 'thing' it is, it needs rectifying.  Both your version and mine display the same so I'm leaning towards skin issue, but I need to dig around in the skin files to see if that really is the case.

Reply:
Possibly both a skin and a core issue. I know that my node hack has an issue with setting item list name (where what gets displayed should come from), but the skin could be doing its own thing on top of that especially if it varies with view type.